Story 00 juksilve

DrJukka opened the pull request:
Coordinator server + performance tests

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/thaliproject/thali_cordovaplugin/128)
<!-- Reviewable:end -->

tobybrad commented:
General comments:

The code looks strait forward and clean. I'm a little confused about the sequence of changes though. Have you manually copied files in from other branches ? I also don't see how you got from the switch/case style of executing tests to the current one.

The readme needs cleaning up and making clearer. I really had to concentrate and fill in a lot of gaps to figure out how to run this. First time I ran index.js I had an error regarding a missing perf test config.

tobybrad commented:
I'd really like to see all the files collected into a single module. Right now they're scattered throughout the jxcore/ and TestServer/ directories. They're not designed to work independently so let's have them as part of a single cohesive module. Node naming convention for modules is lower case also, classes within the module may be camel-cased.

---

Reviewed 31 of 31 files at r1.
Review status: all files reviewed at latest revision, 37 unresolved discussions, all commit checks successful.

---

<sup>**[test/README.md, line 2 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIJ85ZR2SUuHuRLW4V)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/README.md#L2)):</sup>
Proof read the whole of this.

---

<sup>**[test/README.md, line 32 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIJIivHzubyCbWnZT4)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/README.md#L32)):</sup>
Don't really see the need to expose all this extra configuration stuff.. dataAmount and such like. They all should have sensible defaults and the explanations for how to tweak them be local to the test. We're very likely end up with many more of these tests and it doesn't make sense to me to document them in a readme.

---

<sup>**[test/TestServer/config.json, line 1 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIJoL3MkK35JAaINbJ)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/config.json#L1)):</sup>
performance tests doesn't seem to be what testThaliEmitter is.

---

<sup>**[test/TestServer/ipaddress.json, line 1 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIKCKm8aZPwxjdLxCw)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/ipaddress.json#L1)):</sup>
I'd like it better is this was serveraddress.json.

---

<sup>**[test/TestServer/IPAddressToFile.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIKIQxpx2OL1L1KYXo)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/IPAddressToFile.js#L4)):</sup>
Don't get this file at all. I can see what it does.. but why not just 'ifconfig' ??

---

<sup>**[test/TestServer/TestFramework.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIL9oW_kE7rLkNQNDI)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/TestFramework.js#L4)):</sup>
It's not clear to me how this is different to PerfTestFramework. Appears to be doing a lot of the same work.

---

<sup>**[test/TestServer/UnitTestFramework.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzILHr-ACujxyVt3ZmD)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/UnitTestFramework.js#L4)):</sup>
A few comments would be welcome, otherwise... nice.

---

<sup>**[test/www/js/thali_main.js, line 38 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI3mLbOmwRjZ3VTor_)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/js/thali_main.js#L38)):</sup>
This'll play havoc with any CI system. Should be logged and terminate.

---

<sup>**[test/www/jxcore/app.js, line 20 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzILwcX3db7Qixxwgwn)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/app.js#L20)):</sup>
I think we can now just lose the echo server now we have co-ordinated tests.

---

<sup>**[test/www/jxcore/app.js, line 56 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI4CVBoPhqFR03befV)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/app.js#L56)):</sup>
We can lose this block once we have coordinated tests, was only required to avoid races in testNativeLayer.

---

<sup>**[test/www/jxcore/TestFrameworkClient.js, line 41 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI5-UqupqxZXKXevzK)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/TestFrameworkClient.js#L41)):</sup>
start now

---

<sup>**[test/www/jxcore/tests/ReConnectConnector.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI6tP6LOfZo_NjQLzn)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectConnector.js#L4)):</sup>
I don't really understand what this file is for. It *looks* like it's some part of perf test since it appears to be calling connect and disconnect of peers which suggest to me it's an app level perf test. On further inspection it looks like it 1) establishes a connection to a remote peer 2) rapidly spins socket connections to the local socket server Can we have a couple of sentences at the top of every class/file to explain what this thing is for.

---

<sup>**[test/www/jxcore/tests/ReConnectConnector.js, line 84 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI7RGiCFL7CzOKMdEb)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectConnector.js#L84)):</sup>
pedantry: You'e not connected to any port at this point. Port is where you should connect your client socket to have it bridged to the remote peer.

---

<sup>**[test/www/jxcore/tests/ReConnectConnector.js, line 95 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI7x9_kyfrEzCDX2LN)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectConnector.js#L95)):</sup>
No error handling for failure to connect.

---

<sup>**[test/www/jxcore/tests/ReConnectConnector.js, line 97 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIA1ajkIs83Oyf3iFN)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectConnector.js#L97)):</sup>
You probably meant to record numbers so you could check you were echoed what you sent, right ? You're not doing it though.

---

<sup>**[test/www/jxcore/tests/ReConnectConnector.js, line 148 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI5ielg0K5z0G6aYOq)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectConnector.js#L148)):</sup>
socket.end *will* send FIN packet it'd be a bug if the server didn't quickly return a FIN ACK and the socket should emit 'closed'. If 'data just stops flowing' I'd suggest that's actually a bug in our code and not node.net (allowHalfOpen == false by default).

---

<sup>**[test/www/jxcore/tests/ReConnectConnector.js, line 174 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI8pRKpOO20XOzgYX6)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectConnector.js#L174)):</sup>
The connection to the remote peer ? It shouldn't. The local<->local socket, yes.

---

<sup>**[test/www/jxcore/tests/ReConnectTCPServer.js, line 6 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIBIA5FzACYGA-Tlq8)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectTCPServer.js#L6)):</sup>
I think this would be neater if collected into a module with it's client pair.

---

<sup>**[test/www/jxcore/tests/ReConnectTCPServer.js, line 33 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIBBsIJjdweEkwdW-4)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectTCPServer.js#L33)):</sup>
We can probably lose the following if we never intend to actually use it

---

<sup>**[test/www/jxcore/tests/ReConnectTCPServer.js, line 46 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIARJfghx5mdyVbcbo)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectTCPServer.js#L46)):</sup>
No error handling for failure to listen.

---

<sup>**[test/www/jxcore/tests/SendDataConnector.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIBQdVLVLOvIyYcAfW)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/SendDataConnector.js#L4)):</sup>
This is another flavour of perf test, right ? Can we collect these together into a module or at the very least a directory.

In fact, this looks awfully like a dupe of a file I've already reviewed. There's certainly a lot the same. 

---

<sup>**[test/www/jxcore/tests/SendDataTCPServer.js, line 7 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIBzy67on4NUTjuyAf)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/SendDataTCPServer.js#L7)):</sup>
Again.. this looks an awful lot like a dupe of a previous file. Can we have some explanation of how, if at all, this differs from the ReConnect server and if it doesn't let's lose one or the other.

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 5 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIDXvdSp-OOCoIjZHH)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testFindPeers.js#L5)):</sup>
Is this intended to be a sanity check of the coordinator or a full-blown test ?

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 43 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzID0558aMQb0i1apA2)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testFindPeers.js#L43)):</sup>
isn't this loop redundant ? Why not look up directly in the map ? At least break once you've found the peer you were looking for.

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 50 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIE620KR6KS1EDX4Vt)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testFindPeers.js#L50)):</sup>
You're swapping this and self. Pick one and stick to it. Also.. and this may just be my ignorance of node/js inheritance.. but you don't appear to have messed with any prototypes or defined weAreDoneNow() on self.

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 103 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIE1K_8aZsjnpGiCz7)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testFindPeers.js#L103)):</sup>
You're swapping this and self. Pick one and stick to it.

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 106 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIDNURbqyaPlaqW5iM)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testFindPeers.js#L106)):</sup>
Why not 'stopped' ?

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 111 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzICZ4gQU6U203bEy3T)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testFindPeers.js#L111)):</sup>
I'll come out and say it.. I really don't like cutesy function names like 'weAreDoneNow'. Something more like 'finished' suits me much better.

---

<sup>**[test/www/jxcore/tests/testReConnect.js, line 5 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIEbmncqMMIEfnzQni)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testReConnect.js#L5)):</sup>
Ah.. it starts to become clear how these hang together. Can we have some directory structure around this unless we're just looking at hundreds of files in a dir with no obvious relationship between them. I really don't want to have to read every file in a dir to understand structure.

---

<sup>**[test/www/jxcore/tests/testReConnect.js, line 78 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIFLd-BzymQMaDlZ7T)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testReConnect.js#L78)):</sup>
And what are you going to do about it ?? Wait for a timeout I guess ?? That could be a tedious business. No way to quickly fail and move on ??

---

<sup>**[test/www/jxcore/tests/testSendData.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIFmRQuHczRqeC_FXN)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testSendData.js#L4)):</sup>
Again we could do with some structure around these tests. Also.. I'm not wholly convinced this is different enough from ReConnect to require a whole other test. If it *is* then perhaps some of the large amount of commonality could be hoisted into a base class ??

---

<sup>**[test/www/jxcore/tests/testSendData.js, line 57 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIFhRthBX6BVlH09mu)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testSendData.js#L57)):</sup>
Looks like a debug log we could do without.

---

<sup>**[test/www/jxcore/thali-tape.js, line 11 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIGaQTeHB4tC9KH2q-)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/thali-tape.js#L11)):</sup>
Why not UnitTest. Seems like a pointless abbreviation. Also.. why the random number.. doesn't look very repeatable 

---

<sup>**[test/www/jxcore/thali-tape.js, line 16 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIH7-7FU1emZe-h0ax)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/thali-tape.js#L16)):</sup>
Instances have camel-case names. Classes start with capital. 

I'd be setting setMaxListeners(0) on coordinator since you're going to warnings about exceeding number of event listeners since it looks like you're about to add two listeners for every test you're about run. 

---

<sup>**[test/www/jxcore/thali-tape.js, line 41 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIHyhFwR5Foi2dllXv)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/thali-tape.js#L41)):</sup>
Is it not an error to get connect more than once ? What happens if you get disconnected then reconnected ?

---

<sup>**[test/www/jxcore/thali-tape.js, line 47 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIHmYsiEXXmdjx5iN5)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/thali-tape.js#L47)):</sup>
what's _t ? Also.. tearDown not teardDown (perhaps).

---

<sup>**[test/www/jxcore/thali-tape.js, line 70 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIJ-uxHuZAS7KC0uTL)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/thali-tape.js#L70)):</sup>
https://www.npmjs.com/package/once may help you out for this kind of thing. Maybe.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128)
<!-- Sent from Reviewable.io -->

DrJukka commented:
Review status: all files reviewed at latest revision, 32 unresolved discussions, all commit checks successful.

---

<sup>**[test/README.md, line 2 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIJ85ZR2SUuHuRLW4V)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/README.md#L2)):</sup>
I modified the readme according several suggestions made, will likely need native speaker for final proofreading

---

<sup>**[test/README.md, line 32 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIJIivHzubyCbWnZT4)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/README.md#L32)):</sup>
moved to the PerfTestFramework.js, where there is also sull example of the config file's content

---

<sup>**[test/TestServer/config.json, line 1 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIJoL3MkK35JAaINbJ)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/config.json#L1)):</sup>
Actually, this is left here by accident, and also the actual Config_PerfTest.json is missing, now fixed.

---

<sup>**[test/TestServer/ipaddress.json, line 1 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIKCKm8aZPwxjdLxCw)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/ipaddress.json#L1)):</sup>
Done.

---

<sup>**[test/TestServer/IPAddressToFile.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIKIQxpx2OL1L1KYXo)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/IPAddressToFile.js#L4)):</sup>
What do you mean by 'ifconfig' ? As said when I started this part of the project that I have not worked with node before really, and its been long time since I did javascript, so any suggestions that make code better are welcome.

---

<sup>**[test/TestServer/TestFramework.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIL9oW_kE7rLkNQNDI)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/TestFramework.js#L4)):</sup>
Sorry, that file should have been deleted, since it was indeed renamed to PerfTestFramework, after I added the UnitTestFramework.

---

<sup>**[test/TestServer/UnitTestFramework.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzILHr-ACujxyVt3ZmD)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/UnitTestFramework.js#L4)):</sup>
I'll go through the codes today and will add comments

---

<sup>**[test/www/js/thali_main.js, line 38 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI3mLbOmwRjZ3VTor_)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/js/thali_main.js#L38)):</sup>
Ok, that way was generally used to notify problems with app.js file when we have Person running, so will be changing that.

---

<sup>**[test/www/jxcore/app.js, line 20 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzILwcX3db7Qixxwgwn)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/app.js#L20)):</sup>
I suppose the tests might still be requiring this. I did not change them to use any other server, basically just changed tape, to be using thali-tape. As well as moved the stopBroadcasting to be handled on teaerdown synchronized by coordinator server. 

Also there is the debug logging into the screen (which can in sometimes be handy).

---

<sup>**[test/www/jxcore/app.js, line 56 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI4CVBoPhqFR03befV)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/app.js#L56)):</sup>
Done.

---

<sup>**[test/www/jxcore/TestFrameworkClient.js, line 41 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI5-UqupqxZXKXevzK)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/TestFrameworkClient.js#L41)):</sup>
Done.

---

<sup>**[test/www/jxcore/tests/ReConnectConnector.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI6tP6LOfZo_NjQLzn)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectConnector.js#L4)):</sup>
its actually part of the logic for testReConnect.js tests. I'll go through the codes today and add comments to all of them.

---

<sup>**[test/www/jxcore/tests/ReConnectConnector.js, line 95 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI7x9_kyfrEzCDX2LN)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectConnector.js#L95)):</sup>
How would you suggest fixing it ? Basically I never seen that failing, if something happens, I do get the error event handler called for the socket though.

---

<sup>**[test/www/jxcore/tests/ReConnectConnector.js, line 97 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIA1ajkIs83Oyf3iFN)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectConnector.js#L97)):</sup>
Nope, I basically just wanted to make a buffer of certain length, and did not figure any easier way on doing it. Any nicer looking or easier way could always be suggested.

---

<sup>**[test/www/jxcore/tests/ReConnectConnector.js, line 148 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI5ielg0K5z0G6aYOq)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectConnector.js#L148)):</sup>
The end did not work. I basically tried it. As far as I know it does not properly close the socket, and if the server would continue sending something, it would be coming through.

Though, I did now change it to use "self.clientSocket.destroy()", which actually should do the trick. I'll be doing some testing today to see that it actually does work as expected.

---

<sup>**[test/www/jxcore/tests/ReConnectConnector.js, line 174 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI8pRKpOO20XOzgYX6)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectConnector.js#L174)):</sup>
It should. If the socket is properly closed, then the native layer will get notified of it (with error on reading actually), and it will be then doing full and clean disconnect. As we don't have the disconnect specified as proper API (its there, but should not be used, ask Yaron for better explanation), the disconnect is handled by closing the connected socket.

---

<sup>**[test/www/jxcore/tests/ReConnectTCPServer.js, line 33 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIBBsIJjdweEkwdW-4)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectTCPServer.js#L33)):</sup>
Done.

---

<sup>**[test/www/jxcore/tests/ReConnectTCPServer.js, line 46 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIARJfghx5mdyVbcbo)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/ReConnectTCPServer.js#L46)):</sup>
How would you suggest doing & handling it ?
I actually never seen it failing, and I do get later error reported via events.

---

<sup>**[test/www/jxcore/tests/SendDataConnector.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIBQdVLVLOvIyYcAfW)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/SendDataConnector.js#L4)):</sup>
This is the test 3, where we send the big amount of data, where as the re-connect was the test case nr.2, in which we just do multiple re-connections, and see that it works with small amount of data.

---

<sup>**[test/www/jxcore/tests/SendDataTCPServer.js, line 7 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIBzy67on4NUTjuyAf)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/SendDataTCPServer.js#L7)):</sup>
I'll add the comments. Basically the replying is different, the sender just expects to get notified every N-byte of data, and not just being echoed the data send over .

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 5 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIDXvdSp-OOCoIjZHH)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testFindPeers.js#L5)):</sup>
This is the test nr. 1. In which we check how fast we can find peers, needed to make sure how different technologies do the task.

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 43 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzID0558aMQb0i1apA2)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testFindPeers.js#L43)):</sup>
if you have better way on doing this, I would be interested on hearing, as said I'm still on learning phase for node & javascript.

Anyway, the idea is that in the test we need to find certain amount of peers, and I'm here checking on how many we currently think that we see ( peeravailable == true), so I can later on see whether we have reached the needed amount already.

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 50 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIE620KR6KS1EDX4Vt)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testFindPeers.js#L50)):</sup>
I do have sometimes issues with the context in which I'm running different parts with Node, and could always be educated on the issue, so I would be getting it easier way right in the future.

Anyway, this point is run inside the peerAvailabilityChanged callback function, and I want to point into the this of the class, thus I'm using the self variable to do so. I would assume that using this inside the function might actually give me wrong context (the context of the calling party)

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 103 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIE1K_8aZsjnpGiCz7)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testFindPeers.js#L103)):</sup>
I do have sometimes issues with the context in which I'm running different parts with Node, and could always be educated on the issue, so I would be getting it easier way right in the future.

So I'm using this when I'm in the context of the class I'm running, and I use self variable to point to the this of the class when I'm inside a callback that might be called in different context. If I'm wrong with this thinking, I would be really happy  to be corrected.

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 106 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIDNURbqyaPlaqW5iM)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testFindPeers.js#L106)):</sup>
I suppose the naming was supposed to correspond to the fact that weAreDoneNow() being already called, so we would not be calling it again, more of less just preventing us on doing double reporting.

Also we are not stopped in this point yet, we are just done with our stuff, but we are still running and waiting for the coordinator server to tell us when we would be actually stopping the test.

---

<sup>**[test/www/jxcore/tests/testReConnect.js, line 5 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIEbmncqMMIEfnzQni)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testReConnect.js#L5)):</sup>
Yep, I'll add the comments to make it clear on which files belong together. All test**** files are the ones that gets loaded by the framework. All others are just supporting the actual test files.

---

<sup>**[test/www/jxcore/tests/testReConnect.js, line 78 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIFLd-BzymQMaDlZ7T)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testReConnect.js#L78)):</sup>
Currently it just fails, and then timeouts. Though I could of course add code that would inform the coordinator server strait away, so it would not need to wait for the timeout to happen.

---

<sup>**[test/www/jxcore/tests/testSendData.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIFmRQuHczRqeC_FXN)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/testSendData.js#L4)):</sup>
probably could be rather easy to combine them. Though we would need to build a logic to handle the server replying and client getting data parts accordingly. 

---

<sup>**[test/www/jxcore/thali-tape.js, line 11 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIGaQTeHB4tC9KH2q-)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/thali-tape.js#L11)):</sup>
Basically we just need a unique name for a device during the test, so we can determine which device we got data and connections from at the coordinator server.

The name is then not really used in anything else except that its logged into the coordinator server screen.

---

<sup>**[test/www/jxcore/thali-tape.js, line 41 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIHyhFwR5Foi2dllXv)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/thali-tape.js#L41)):</sup>
I added now that disconnect will set the value to false, so then it can be reconnected.

---

<sup>**[test/www/jxcore/thali-tape.js, line 47 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIHmYsiEXXmdjx5iN5)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/thali-tape.js#L47)):</sup>
t is the argument name for tests generally used with tape. so tearDown_t and setUp_t are holding the test variables which need to be Ended once we can move forward. 

Fixed the tearDown_t 

---

<sup>**[test/www/jxcore/thali-tape.js, line 70 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIJ-uxHuZAS7KC0uTL)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/thali-tape.js#L70)):</sup>
I would find the current way easier to track what actually happens, since the both setUp_t  and tearDown_t will be getting set with new values for each test we run.

Also there is the possibility that we get disconnected, and then I'll be calling fail for these.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128)
<!-- Sent from Reviewable.io -->

yaronyg commented:
Review status: 7 of 31 files reviewed at latest revision, 35 unresolved discussions, all commit checks successful.

---

<sup>**[test/TestServer/TestDevice.js, line 2 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzLrwRpFSnjRSLVmKjW)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/82cba9f47b4c5d92e5be7f19e495d5951c4ab42f/test/TestServer/TestDevice.js#L2)):</sup>
/calss/class

---

<sup>**[test/www/jxcore/TestFrameworkClient.js, line 2 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzLskHdilI9n-ns51u3)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/82cba9f47b4c5d92e5be7f19e495d5951c4ab42f/test/www/jxcore/TestFrameworkClient.js#L2)):</sup>
/perfoamnce/performance

---

<sup>**[test/www/jxcore/TestFrameworkClient.js, line 41 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI5-UqupqxZXKXevzK)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/TestFrameworkClient.js#L41)):</sup>
Um... no it's not. It still says "Star"

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 38 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzLtvQdx2ZSqZYbFQRp)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/82cba9f47b4c5d92e5be7f19e495d5951c4ab42f/test/www/jxcore/tests/testFindPeers.js#L38)):</sup>
Wouldn't this code be easier to read if it used peers.foreach?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128)
<!-- Sent from Reviewable.io -->

tobybrad commented:
Reviewed 29 of 29 files at r2.
Review status: all files reviewed at latest revision, 34 unresolved discussions, all commit checks successful.

---

<sup>**[test/TestServer/IPAddressToFile.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIKIQxpx2OL1L1KYXo)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/IPAddressToFile.js#L4)):</sup>
ifconfig's a shell command on unix systems that tells you what your ip address is. ifconfig is the windows equivalent.

---

<sup>**[test/www/jxcore/tests/SendDataConnector.js, line 4 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIBQdVLVLOvIyYcAfW)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/SendDataConnector.js#L4)):</sup>
Sounds a lot like one can just be a reparameterised run of another to me.

---

<sup>**[test/www/jxcore/thali-tape.js, line 11 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIGaQTeHB4tC9KH2q-)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/thali-tape.js#L11)):</sup>
yeah.. so if something goes wrong you're saying there's no way to figure out which devices generated that error ?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128)
<!-- Sent from Reviewable.io -->

DrJukka commented:
Did go through them now, should now be fixed.

---

Review status: 27 of 30 files reviewed at latest revision, 34 unresolved discussions, all commit checks successful.

---

<sup>**[test/README.md, line 24 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIJIivHzubyCbWnZT4)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/README.md#L24)):</sup>
changes made look great :)

---

<sup>**[test/TestServer/IPAddressToFile.js, line 6 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIKIQxpx2OL1L1KYXo)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/TestServer/IPAddressToFile.js#L6)):</sup>
I think its better to have the file, it would also allow us to use server based coordinator server. Basically the file would tell where the server can be found, not just the IP address of the wifi we are using.

---

<sup>**[test/TestServer/TestDevice.js, line 2 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzLrwRpFSnjRSLVmKjW)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/82cba9f47b4c5d92e5be7f19e495d5951c4ab42f/test/TestServer/TestDevice.js#L2)):</sup>
Done.

---

<sup>**[test/www/jxcore/tests/SendDataConnector.js, line 17 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIBQdVLVLOvIyYcAfW)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/tests/SendDataConnector.js#L17)):</sup>
Could, but we would also need to see that server starts replying right, and that we do use the right logic on determining that we got all data, that also requires full testing to be sure its still working right.

---

<sup>**[test/www/jxcore/tests/testFindPeers.js, line 38 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzLtvQdx2ZSqZYbFQRp)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/82cba9f47b4c5d92e5be7f19e495d5951c4ab42f/test/www/jxcore/tests/testFindPeers.js#L38)):</sup>
Done.

---

<sup>**[test/www/jxcore/thali-tape.js, line 29 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzIGaQTeHB4tC9KH2q-)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/thali-tape.js#L29)):</sup>
As I understood, the tape  results are logged on device, and those are used to analyse the results, thus it should be visible there, with unit tests the coordinator server is more-or-less dummy one that just synchronizes stuff. With performance tests it would be great to get something we can identify, but as the get/setName functionality was removed from the API last May, we don't really have a way on uniquely identifying the devices.

All, and all, there is options setting with thali-tape, which could be used for setting the name, thus if we get test-time device names, we could indeed start using it.

---

<sup>**[test/www/jxcore/TestFrameworkClient.js, line 2 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzLskHdilI9n-ns51u3)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/82cba9f47b4c5d92e5be7f19e495d5951c4ab42f/test/www/jxcore/TestFrameworkClient.js#L2)):</sup>
Actually the file is now deleted. I renamed earlier to PerfTestFramework, when I though I need to write one for unit tests as well.

---

<sup>**[test/www/jxcore/TestFrameworkClient.js, line 48 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128#-JzI5-UqupqxZXKXevzK)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/280289fd9fff0c182b7744327c5424d9595061f2/test/www/jxcore/TestFrameworkClient.js#L48)):</sup>
Actually the file is now deleted. I renamed earlier to PerfTestFramework, when I though I need to write one for unit tests as well.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/128)
<!-- Sent from Reviewable.io -->
